### PR TITLE
Don't pass override option if it hasn't been configured

### PR DESCRIPTION
**Pull request by brianhlin**
Currently untested...

### CODE REVIEW

File: 30-remote-site-setup.sh [hunk 1]
```diff
@@ -66,20 +65,24 @@ echo $REMOTE_HOST_KEY >> $root_ssh_dir/known_hosts
...
 users=$(cat /etc/grid-security/grid-mapfile /etc/grid-security/voms-mapfile | \
             awk '/^"[^"]+" +[a-zA-Z0-9\-\._]+$/ {print $NF}' | \
             sort -u)
 [[ -n $users ]] || exit 1
 
+override_opts=""
 for ruser in $users; do
     setup_ssh_config
     setup_endpoints_ini
+    [[ -d $OVERRIDE_DIR ]] && override_opts="-o $OVERRIDE_DIR"
```
Comment (edquist) [inline 30-remote-site-setup.sh]: Small detail here, but i guess there is no warning if `OVERRIDE_DIR` is set, but to a non-existing directory

File: 30-remote-site-setup.sh [hunk 2]
```diff
@@ -66,20 +65,24 @@ echo $REMOTE_HOST_KEY >> $root_ssh_dir/known_hosts
 # Populate the bosco override dir from a Git repo
 GIT_SSH_KEY=/etc/osg/git.key
 [[ -f $GIT_SSH_KEY ]] && export GIT_SSH_COMMAND="ssh -i $GIT_SSH_KEY"
-[[ -z $BOSCO_GIT_ENDPOINT || -z $BOSCO_DIRECTORY ]] || \
+if [[ -z $BOSCO_GIT_ENDPOINT || -z $BOSCO_DIRECTORY ]]; then
```
Comment (edquist) [inline 30-remote-site-setup.sh]: it seems like the logic has reversed here -

looks like it used to be "only run bosco-override-setup.sh if both `BOSCO_GIT_ENDPOINT` and `BOSCO_DIRECTORY` are set;

now it's "run bosco-override-setup.sh if at least one of `BOSCO_GIT_ENDPOINT` or `BOSCO_DIRECTORY` is _unset_"

is this change in the logic intentional?

File: 30-remote-site-setup.sh (hunk 1 above)
Comment (brianhlin) [reply 30-remote-site-setup.sh]: Fixed

File: 30-remote-site-setup.sh (hunk 2 above)
Comment (brianhlin) [reply 30-remote-site-setup.sh]: Good catch, fixed

File: 30-remote-site-setup.sh [hunk 3]
```diff
@@ -78,11 +79,19 @@ grep '^OSG_GRID="/cvmfs/oasis.opensciencegrid.org/osg-software/osg-wn-client' \
      /var/lib/osg/job-environment*.conf > /dev/null 2>&1
 cvmfs_wn_client=$?
 
+override_opts=""
 for ruser in $users; do
     setup_ssh_config
     [[ $cvmfs_wn_client -eq 0 ]] || setup_endpoints_ini
+    if [[ -n $OVERRIDE_DIR ]]; then
```
Comment (edquist) [inline 30-remote-site-setup.sh]: While it's not broken as you have it, i think this new `if` block really belongs above the `for` loop; just to be executed once.

**Pull request merged**